internal/repository/master_index.go

```diff
@@ -29,10 +29,12 @@ func (mi *MasterIndex) Lookup(id restic.ID, tpe restic.BlobType) (blobs []restic
 	debug.Log("looking up id %v, tpe %v", id.Str(), tpe)
 
 	for _, idx := range mi.idx {
-		blobs, err = idx.Lookup(id, tpe)
-		if err == nil {
-			debug.Log("found id %v: %v", id.Str(), blobs)
-			return
+		if idx.Has(id, tpe) {
```

Comment:
This change won't cause any issues, since blobs are only added to an index, so it cannot happen that `Has()` succeeds and subsequent `Lookup()` fails. But now you have the map lookup twice.

I think we can revert this change and do better: Since the only error `Lookup()` ever returns is `not found`, how about we change the function signature like like this:
```go
func (idx *Index) Lookup(id restic.ID, tpe restic.BlobType) (blobs []restic.PackedBlob, found bool) {
```

That would mean touching every place where `Lookup()` is called, but shouldn't be too hard to do.

Now the lookup is done only once, and there's no error built in any case.

What do you think?

Reply:
Sounds good to me, I'll take a look into doing that this weekend.  Hopefully I'll have some good news/better benchmarks!